remove_wheels.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# fail on undefined variables
+set -u
+# Prevent pipe errors to be silenced
+set -o pipefail
+# Exit if any command exit as non-zero
+set -e
+# enable trace mode (print what it does)
+set -x
+
+# get the anaconda token from the github secrets
+#
+# this is to prevent accidental removals
+echo "Getting anaconda token from github secrets..."
+
+ANACONDA_USER="${INPUT_ANACONDA_USER}"
+ANACONDA_TOKEN="${INPUT_ANACONDA_TOKEN}"
+N_LATEST_UPLOADS="${INPUT_N_LATEST_UPLOADS}"
+
```

Comment:
I am not sure if the Anaconda token that is authorised to maintainers of, say, a package X is restricted to only remove wheels for said package X, or whether it can remove all packages in the index. This is because while users of SPNW won't be affected since we handle the deletions, users with their own organisation with wheels for packages X, Y, and Z being uploaded to it would either:

- run this action somewhere centrally to remove all (but the latest N) uploads of X, Y, and Z; or
- they would want to run the deletions from separate repositories for X, Y, and Z and want to delete just X, Y, and Z at a time on each package's own accord and deletion schedule.

Which situation would be more plausible? If it is the latter, we'd have to provide another input in the action for a comma-separated string of packages to delete uploads for (and maybe `*` for deleting all packages?). To replace the existence of `packages-ignore-from-cleanup.txt` (which would exist only in this repository – see below), it might also make sense to include a whitelist input, too. If it's the former, it gets easier to implement, but users won't have fine-grained control on automations for what packages are being deleted.

P.S. This is all valid only if I'm not missing something about how the index and its permissions are structured :) Happy to receive others' thoughts!

Reply:
Based on @tupui's suggestion in https://github.com/scientific-python/upload-nightly-action/pull/95#issuecomment-2386447252, users will need to run the action multiple times to remove old uploads for multiple package versions. Hence, the second option of allowing per-package deletions is better, and an input for a list of packages to delete wheels for isn't required.

Reply:
However, if someone has an appropriately scoped token for their organisation, they can very well remove multiple wheels from the index at a time, and some users might want to do that. So, it might be necessary to mention in the documentation how the tokens work and pre-emptively warn about possible deletions in an admonition.

Reply:
To be clear, users have full control over their project. So they can add or remove any wheel they want on that project. People can just not add members to their project and they need to ask admins.